common/v2/features/Settings/components/AddressBook.tsx

```diff
@@ -69,37 +74,77 @@ const SEditableText = styled(EditableText)`
   }
 `;
 
-export const screenIsMobileSized = (breakpoint: number): boolean =>
-  window.matchMedia(`(max-width: ${breakpoint}px)`).matches;
-
 export default function AddressBook({
   addressBook,
+  addressBookRestore,
   toggleFlipped,
   deleteAddressBooks,
-  updateAddressBooks
+  updateAddressBooks,
+  restoreDeletedAddressBook
 }: Props) {
   const [deletingIndex, setDeletingIndex] = useState();
+  const [undoDeletingIndexes, setUndoDeletingIndexes] = useState<[number, TUuid][]>([]);
+  const overlayRows: [number[], [number, TUuid][]] = [
+    isNumber(deletingIndex) ? [deletingIndex] : [],
+    [...undoDeletingIndexes]
+  ];
+  const overlayRowsFlat = [...overlayRows[0], ...overlayRows[1].map(row => row[0])];
 
-  const overlayRows = [deletingIndex];
+  const getDisplayAddressBook = (): ExtendedAddressBook[] => {
+    const accountsTemp = cloneDeep(addressBook);
+    overlayRows[1]
+      .sort((a, b) => a[0] - b[0])
+      .forEach(index => {
+        accountsTemp.splice(index[0], 0, addressBookRestore[index[1]] as ExtendedAddressBook);
+      });
+    return accountsTemp.sort((a, b) => a.uuid.localeCompare(b.uuid));
+  };
+  const displayAddressBook = getDisplayAddressBook();
 
   const addressBookTable = {
     head: ['Favorite', 'Label', 'Address', 'Network', 'Notes', 'Delete'],
-    overlay:
-      overlayRows && overlayRows[0] !== undefined ? (
-        <RowDeleteOverlay
-          prompt={`Are you sure you want to delete ${addressBook[overlayRows[0]].label} address with
-             address: ${truncate(addressBook[overlayRows[0]].address)}?`}
-          deleteAction={() => {
-            deleteAddressBooks(addressBook[overlayRows[0]].uuid);
-            setDeletingIndex(undefined);
-          }}
-          cancelAction={() => setDeletingIndex(undefined)}
-        />
-      ) : (
-        <></>
-      ),
-    overlayRows,
-    body: addressBook.map(
+    overlay: (rowIndex: number): JSX.Element => {
+      if (overlayRows) {
+        if (overlayRows[0].length && overlayRows[0][0] === rowIndex) {
+          // Row delete overlay
+          const { uuid, label, address } = displayAddressBook[rowIndex];
```

Comment:
Should be able to move this above the if-statement too. Also try not to nest conditionals. Use inverse first:
```
if (!overlayRows) return (<></>);

const { uuid, label, address } = displayAddressBook[rowIndex];
if (overlayRows[0].length && overlayRows[0][0] === rowIndex) { ...
```